microapps/PopupApp/app/src/main/java/com/arcgismaps/toolkit/popupapp/screens/mapscreen/MainScreen.kt

```diff
@@ -63,13 +68,28 @@ fun MainScreen(viewModel: MapViewModel) {
             skipHiddenState = false
         )
     )
+    LaunchedEffect(scaffoldState.bottomSheetState.currentValue) {
+        if (scaffoldState.bottomSheetState.currentValue == SheetValue.Hidden) {
+            resetSelection(viewModel)
+        }
+    }
+
     BottomSheetScaffold(
         sheetContent = {
-            if (viewModel.popup != null) {
-                Popup(viewModel.popup!!, Modifier.animateContentSize())
+            if (viewModel.popupState != null) {
+                Popup(
+                    viewModel.popupState!!,
```

Comment:
could use smart casting here rather than the !! operator.

```suggestion
            val state = viewModel.popupState
            if (state != null) {
                Popup(
                    state,
```